test/appender/test_appender.cpp

```diff
@@ -571,3 +571,187 @@ TEST_CASE("Test appending to different database files", "[appender]") {
 	REQUIRE(failed);
 	REQUIRE_NO_FAIL(con.Query("COMMIT TRANSACTION"));
 }
+
+void setDataChunkInt32(DataChunk &chunk, idx_t col_idx, idx_t row_idx, int32_t value) {
+	auto &col_i = chunk.data[col_idx];
+	auto col_i_data = reinterpret_cast<int32_t *>(col_i.GetData());
+	col_i_data[row_idx] = value;
+}
+
+TEST_CASE("Test appending with an active default column", "[appender]") {
+	duckdb::unique_ptr<QueryResult> result;
+	DuckDB db(nullptr);
+	Connection con(db);
+
+	REQUIRE_NO_FAIL(con.Query("CREATE OR REPLACE TABLE tbl (i INT DEFAULT 4, j INT, k INT DEFAULT 30)"));
+	Appender appender(con, "main", "tbl");
+	appender.AddColumn("i");
+
+	DataChunk chunk;
+	const duckdb::vector<LogicalType> types = {LogicalType::INTEGER};
+	chunk.Initialize(*con.context, types);
+
+	setDataChunkInt32(chunk, 0, 0, 42);
+	setDataChunkInt32(chunk, 0, 1, 43);
+
+	chunk.SetCardinality(2);
+	appender.AppendDataChunk(chunk);
+	appender.Close();
+
+	result = con.Query("SELECT i, j, k FROM tbl");
+	REQUIRE(CHECK_COLUMN(result, 0, {42, 43}));
+	REQUIRE(CHECK_COLUMN(result, 2, {30, 30}));
+
+	// Test that column 1 is NULL.
+	auto result_chunk = result->Fetch();
+	auto &null_result_vec = result_chunk->data[1];
+	UnifiedVectorFormat result_format;
+	null_result_vec.ToUnifiedFormat(result_chunk->size(), result_format);
+	REQUIRE(!result_format.validity.RowIsValid(0));
+	REQUIRE(!result_format.validity.RowIsValid(1));
+}
+
+TEST_CASE("Test appending with an active normal column", "[appender]") {
+	duckdb::unique_ptr<QueryResult> result;
+	DuckDB db(nullptr);
+	Connection con(db);
+
+	REQUIRE_NO_FAIL(con.Query("CREATE OR REPLACE TABLE tbl (i INT DEFAULT 4, j INT, k INT DEFAULT 30)"));
```

Comment:
Can we add a test for a non-foldable default value (e.g. `nextval('seq')` or `random()`)?